### ISSUE TITLE
Replace FolderID by FolderUID in API for Grafana v9

### ISSUE THREAD

**Issue by ying-jeanne** [grafana/grafana#48334]
`FolderId` without a  matching `FolderUid` can be found in `models.Dashboard` and `models.DashboardAclInfoDTO`.
The prior is not used as a DTO and is enhanced at the API level to include FolderUid.

**Comment by zserge** [comment]
`models.Dashboard` is not used as a DTO and is wrapped into the other struct at the API level, which injects FolderUID at runtime for all folders with ID > 0.

`models.DashboardAclInfoDTO` contains FolderID field, which is mutually exclusive with DashboardID. When permissions are requested for a dashboard - DashboardID + UID are set, when permissions are requested for a folder - FolderID + UID are set.

At a guardian level IDs are used (and not UID), but those are hidden from the API level.

In other words, there is nothing left to do on this topic from the API perspective.
